components/markdown/src/markdown.rs

```diff
@@ -485,7 +494,8 @@ pub fn markdown_to_html(
                     });
                 }
                 Event::Html(text) => {
-                    if text.contains("<!-- more -->") {
+                    dbg!(&text);
+                    if MORE_DIVIDER_RE.is_match(&text) {
```

Comment:
you should check first if `has_summary` is already set to true, otherwise you might push multiple continue reading event